lib/neo4j/active_node/query/query_proxy_enumerable.rb

```diff
@@ -14,6 +14,8 @@ def each(node = true, rel = nil, &block)
         end
 
         def result(node = true, rel = nil)
+          return [].freeze if @start_object && @start_object.new_record?
```

Comment:
as I remember this will not help to prevent querying the database on updates in active model hooks which should use the newly updated values instead of the old values from the database.